src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -458,11 +461,17 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
                     case TypeKind.Interface:
                         foreach (var t in localInterfaces)
                         {
-                            if (TypeSymbol.Equals(t, baseType, TypeCompareKind.ConsiderEverything))
+                            if (t.Equals(baseType, TypeCompareKind.ConsiderEverything))
                             {
                                 diagnostics.Add(ErrorCode.ERR_DuplicateInterfaceInBaseList, location, baseType);
                                 continue;
                             }
+                            else if (t.Equals(baseType, TypeCompareKind.ObliviousNullableModifierMatchesAny))
+                            {
+                                // duplicates with ?/! differences are reported later, we report local differences between oblivious and ?/! here
+                                diagnostics.Add(ErrorCode.WRN_DuplicateInterfaceWithNullabilityMismatchInBaseList, location, baseType, this);
+                                continue;
```

Comment:
>continue; [](start = 32, length = 9)

Do we want to continue here though? This will drop the interface from the list. How do we decide which one is better to keep? Perhaps we should report a warning and keep it? Or report a different warning that would make it clear that the thing is ignored?

Reply:
This `continue` is scoped to `foreach`. It doesn't affect the list of interfaces that are collected.
Added test to illustrate and removed both `continue` statements to improve readability.

---
In reply to: [325899242](https://github.com/dotnet/roslyn/pull/38563#discussion_r325899242) [](ancestors = 325899242)

Reply:
Thanks, my mistake

---
In reply to: [326305398](https://github.com/dotnet/roslyn/pull/38563#discussion_r326305398) [](ancestors = 326305398,325899242)